### PR TITLE
Milestone: Budget app

### PR DESCRIPTION
**Pull request by Kashif-Saleem-Ghuman**
Hi Code Reviewer 👋 

🟢 In this milestone

- [ ] Create a Ruby on Rails application that allows the user to:

  - [ ] register and log in, so that the data is private to them.
  - [ ] introduce new transactions associated with a category.
  - [ ] see the money spent on each category.
  - [ ] Transactions list page
 
- [ ] Project requirements
  - [ ] Design
    - [ ] You should follow these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:
    - [ ] Colors.
    - [ ] Typography: font face, size and weight.
    - [ ] Layout: composition and space between elements.

  - [ ] Interactions
    - [ ] Splash screen 
      - [ ] A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.
      - [ ] Sign up and log in pages
      - [ ] The user should be able to register in the app with full name, email and password (all mandatory).
      - [ ] The user can log into the app using email and password.
      - [ ] If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).
    - [ ] Home page (categories page)
      - [ ] When the user logs in, they are presented with the categories page.
      - [ ] For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
      - [ ] When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
      - [ ] There is a button "add a new category" at the bottom that brings the user to the page to create a new category.
    - [ ] Transactions page
      - [ ] For a given category, the list of transactions is presented, ordered by the most recent.
      - [ ] At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
      - [ ] There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
      - [ ] When the user clicks on the "Back" button (<), the user navigates to the home page.
    - [ ] "Add a new category" page
      - [ ] The user fills out a form to create a new category, indicating their name and icon (both mandatory).
      - [ ] The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
      - [ ] When the user clicks on the "Back" button (<), the user navigates to the home page.
    - [ ] "Add a new transaction" page
      - [ ] The user fills out a form to create a new transaction with:
      - [ ] name (mandatory)
      - [ ] amount (mandatory)
      - [ ] categories (mandatory at least one)
      - [ ] The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
      - [ ] When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.
    - [ ] Testing requirements
      - [ ] Create unit and integration tests for all the most important components of your RoR application.
    - [ ] Technical requirements
      - [ ] You should use Postgres as your database.
      - [ ] You should use devise for authentication.
      - [ ] You should validate all user input to make sure that anyone with bad intentions cannot compromise your app.
      - [ ] You can use a view template engine of your choice (.erb, .slim, .haml).
      - [ ] The project should be deployed and accessible online.
      - [ ] Your database schema should reflect the following structure:
[Project presentation Video](https://www.loom.com/share/c318961c3875475d83986dd2b0a1532e)(updated)

### CODE REVIEW

File: app/views/groups/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    <div class="card-custom my-2 p-2">
+      <div class="card-body">
+        <div class="d-flex align-items-center">
+          <% if group.icon.attached? %>
+            <div>
+              <%= image_tag url_for(group.icon), class: "img" %>
+            </div>
+          <% end %>
+          <div class="d-flex flex-column ms-3">
+            <div class="d-flex justify-content-between align-items-center w-100 gap-2">
```
Comment (BenMukebo) [inline app/views/groups/index.html.erb:19]: - Kindly improve the design, Check these [design guidelines](https://www.behance.net/gallery/19759151/Snapscan-iOs-design-and-branding?tracking_source=), including:
* Colors
* Typography: font face, size, and weight.
* Layout: composition and space between elements.
- [x] Kindly follow the provided design (width, font, size, height, background, color...)
1. The `ADD NEW CATEGORY` button must always be at the bottom of your page. ⬇️  
2. Add this white border around the ADD NEW CATEGORY button text. 👍🏻 
3. The `ADD NEW PURCHASE` button has a similar problem. Kindly fix it as well. 💯  👍🏻
4. The total amount number should be bold and the text should be normal, Please increase this font width. 👍🏻
5. The card should have a category image, name, `created date`, and amount. kindly let's display the create date for each category card. 🛀🏻 
6. Please respect the height properties for each category or group card.
7. I don't think it is necessary to have two hamburger menu buttons, Kindly remove one 🚮 

Your Implementation | Original Design
:---:|:---:
image| image

File: spec/views/groups_index_spec.rb [hunk 2]
```diff
@@ -0,0 +1,31 @@
+require 'rails_helper'
+
+RSpec.describe 'categories index', type: :feature do
+  describe 'Index' do
+    before(:each) do
+      User.destroy_all
+      Group.destroy_all
+      @u1 = User.new(name: 'Anna', password: '123456', email: 'anna@gmail.com')
```
Comment (BenMukebo) [inline spec/views/groups_index_spec.rb:8]: - [ ] Great job so far 👏🏻 However according to the project requirements. The App should have unit tests and integration tests for all the most important components of the RoR application. Kindly make sure all the components have `integration` tests ✍🏻

**Review by BenMukebo — CHANGES_REQUESTED** [review]
## Hi @Kashif-Saleem-Ghuman, 

You did a great job so far 👏🏼 👏🏼
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


### To Highlight 🎉
- Professional README file. :heavy_check_mark:
- PR has a good title and summary. ✔️
- Good presentation video. ✔️ 
- Your app has been well deployed to render. ✔️ 
- The database schema reflect the ERD diagram. ✔️ 
- Associations between models has been setup in a correct way. ✔️
- The user input has validations. ✔️

## STATUS: CHANGES REQUESTED  ♻️ 
_Please Check the comments under the review and make all required changes_
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers :clinking_glasses:  and Happy coding!!! :dancers:💻  

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please ping me @BenMukebo when you comment so I can receive the notification.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Kashif-Saleem-Ghuman** [comment]
@BenMukebo Thank you for reviewing and providing valuable feedback. I will implement the changes as per the project requirements.

However, you required 'The card should have a category image, name, created date, and amount. kindly let's display the creation date for each category card. 🛀🏻'
 
Whereas the project requirement is 'For each category, the user can see their name, icon and the total amount of all the transactions that belong to that category.'

The created date is not part of the project requirements.

**Comment by BenMukebo** [comment]
Hello @Kashif-Saleem-Ghuman, alright Let me make this one as an optional requested. 👍🏻

**Review by FrederickMih — APPROVED** [review]
## Hi :raising_hand:  @Kashif-Saleem-Ghuman , 

In my opinion, your project is ready for presentation!  There is nothing else to say other than... it's time to merge it :shipit: 

## Highlights :green_circle: 

- PR is professional :+1: 
- App followed the provided design guidelines :+1: 
- Linter checks passed :heavy_check_mark: 

## Optional suggestions
**N/A**


Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Comment by Kashif-Saleem-Ghuman** [comment]
@FrederickMih Thank you very much for approving and appreciating!

**Pull request merged**